bot/src/bot.rs

```diff
@@ -39,28 +40,58 @@ pub fn get_provider(config: &Config) -> SequencerGatewayProvider {
     }
 }
 
+fn to_uint256(n: BigInt) -> (FieldElement, FieldElement) {
```

Comment:
This one is a bit expensive. Let's try to optimize it a little bit.

/ and % both use the same algorithm, we can divide the computation task by two by using ``div_rem``. Also converting to string and back to a FieldElement is slow, loading the bytes directly should be must cheaper. Finally Rust is smart and optimizes computation for stuff like 2^128 at compilation for primitive types, but I don't think it works for BigInt. Using lazy_static should help.
I suggest something like:
```rs

lazy_static! {
    static ref TWO_POW_128: BigInt = BigInt::from(2).pow(128);
}

fn to_uint256(n: BigInt) -> (FieldElement, FieldElement) {
    let (n_high, n_low) = n.div_rem(TWO_POW_128);
    let (_, low_bytes) = n_low.to_bytes_be();
    let (_, high_bytes) = n_high.to_bytes_be();

    (
        FieldElement::from_byte_slice_be(&low_bytes).unwrap(),
        FieldElement::from_byte_slice_be(&high_bytes).unwrap(),
    )
}
```

You'll need to ``cargo add num_integer``, a dependency of BigInt.